### PR TITLE
Add `iotedge config mp` to create a super-config with a manual-provisioning connection string.

### PR DESCRIPTION
**Pull request by arsing**
Other changes:

- Define `IOTEDGE_{CONNECT,LISTEN}_{MANAGEMENT,WORKLOAD}_URI` env vars
  at compile time so that code can know these target-OS-specific values.
  Previously the `IOTEDGE_HOST` env var was used just for the value of
  `IOTEDGE_CONNECT_MANAGEMENT_URI`.

- Implement `Default` for `edgelet_core::{Connect,Listen,MobyRuntime}`
  and make them not-required in the super-config template. Also make the `agent`
  property optional and hard-code its default in the code.

### CODE REVIEW

File: edgelet/Makefile [hunk 1]
```diff
@@ -78,6 +86,10 @@ distclean:
 	rm -rf .cargo
 	rm -rf $(vendordir)
 
+deb: CONNECT_MANAGEMENT_URI=unix:///var/run/iotedge/mgmt.sock
+deb: CONNECT_WORKLOAD_URI=unix:///var/run/iotedge/mgmt.sock
```
Comment (gordonwang0) [inline edgelet/Makefile:90]: This should be `workload.sock`.

File: edgelet/edgelet-core/src/settings.rs [hunk 2]
```diff
@@ -27,6 +27,34 @@ impl Connect {
...
+    #[allow(clippy::option_if_let_else)]
+    fn default() -> Self {
+        const DEFAULT_MANAGEMENT_URI: &str =
+            if let Some(value) = option_env!("IOTEDGE_CONNECT_MANAGEMENT_URI") {
+                value
+            } else {
+                "unix:///var/run/iotedge/mgmt.sock"
+            };
+        const DEFAULT_WORKLOAD_URI: &str =
+            if let Some(value) = option_env!("IOTEDGE_CONNECT_MANAGEMENT_URI") {
```
Comment (gordonwang0) [inline edgelet/edgelet-core/src/settings.rs:41]: This should be `IOTEDGE_CONNECT_WORKLOAD_URI`.

File: edgelet/edgelet-core/src/settings.rs [hunk 3]
```diff
@@ -49,6 +77,35 @@ impl Listen {
...
+    #[allow(clippy::option_if_let_else)]
+    fn default() -> Self {
+        const DEFAULT_MANAGEMENT_URI: &str =
+            if let Some(value) = option_env!("IOTEDGE_CONNECT_MANAGEMENT_URI") {
+                value
+            } else {
+                "fd://aziot-edged.mgmt.socket"
+            };
+        const DEFAULT_WORKLOAD_URI: &str =
+            if let Some(value) = option_env!("IOTEDGE_CONNECT_MANAGEMENT_URI") {
```
Comment (gordonwang0) [inline edgelet/edgelet-core/src/settings.rs:91]: This should be `IOTEDGE_CONNECT_WORKLOAD_URI`.

**Review by gordonwang0 — CHANGES_REQUESTED** [review]

File: edgelet/Makefile (hunk 1 above)
Comment (arsing) [reply edgelet/Makefile:90]: Fixed.

File: edgelet/edgelet-core/src/settings.rs (hunk 2 above)
Comment (arsing) [reply edgelet/edgelet-core/src/settings.rs:41]: Fixed.

File: edgelet/edgelet-core/src/settings.rs (hunk 3 above)
Comment (arsing) [reply edgelet/edgelet-core/src/settings.rs:91]: Fixed.

File: edgelet/iotedge/src/system.rs [hunk 4]
```diff
@@ -13,27 +13,30 @@ use crate::error::{Error, ErrorKind};
...
-        // CentOS 7 doesn't use socket activation.
-        let sockets: &'static [&'static str] = option_env!("IOTEDGE_HOST").map_or(
-            &["aziot-edged.mgmt.socket", "aziot-edged.workload.socket"],
-            |_host| &[],
-        );
+        // If IOTEDGE_LISTEN_MANAGEMENT_URI isn't set at compile-time, assume socket activation is being used.
+        //
+        // This doesn't matter for release builds since those always have IOTEDGE_LISTEN_MANAGEMENT_URI set.
+        // It's only useful for developers' builds.
+        let uses_socket_activation = option_env!("IOTEDGE_LISTEN_MANAGEMENT_URI").map_or(true, |value| value.starts_with("fd://"));
```
Comment (massand) [inline edgelet/iotedge/src/system.rs:20]: It is best to be explicit here and check for cfg_attr(debug_assertions)

Comment (arsing) [reply edgelet/iotedge/src/system.rs:20]: It's not about debug-vs-not. It's about `make rpm` / `make deb` vs not.

Comment (massand) [reply edgelet/iotedge/src/system.rs:20]: I get it now. The comment threw me off as it read "release builds" (I suggest "package build") and "developers' build" (I suggest "non-RPM/DEB builds")

Comment (arsing) [reply edgelet/iotedge/src/system.rs:20]: Oh right. Fixed.

**Review by massand — APPROVED** [review]

**Review by gordonwang0 — APPROVED** [review]

**Review by gordonwang0 — APPROVED** [review]

**Pull request merged**